build-system/common/ci.js

```diff
@@ -225,6 +226,17 @@ function ciBuildSha() {
   return isPullRequestBuild() ? ciPullRequestSha() : ciCommitSha();
 }
 
+/**
+ * Signal to dependent jobs that they should be skipped.
+ *
+ * Currently only relevant for CircleCI builds.
+ */
+function signalGracefulHalt() {
```

Comment:
Since `ci.js` only provides existing state from the environment, I think this function should live in `utils.js`.

Reply:
Done